DeviceDetector.php

```diff
@@ -337,60 +346,39 @@ public function isBot()
      *
      * @return bool
      */
-    public function isTouchEnabled()
+    public function isTouchEnabled(): bool
     {
         $regex = 'Touch';
-        return !!$this->matchUserAgent($regex);
-    }
 
-    /**
-     * Returns if the parsed UA contains the 'Android; Tablet;' fragment
-     *
-     * @return bool
-     */
-    protected function hasAndroidTableFragment()
-    {
-        $regex = 'Android( [\.0-9]+)?; Tablet;';
         return !!$this->matchUserAgent($regex);
     }
 
     /**
-     * Returns if the parsed UA contains the 'Android; Mobile;' fragment
+     * Returns if the parsed UA is detected as a mobile device
      *
      * @return bool
      */
-    protected function hasAndroidMobileFragment()
-    {
-        $regex = 'Android( [\.0-9]+)?; Mobile;';
-        return !!$this->matchUserAgent($regex);
-    }
-
-    protected function usesMobileBrowser()
-    {
-        return $this->getClient('type') == 'browser' && Browser::isMobileOnlyBrowser($this->getClient('short_name'));
-    }
-
-    public function isMobile()
+    public function isMobile(): bool
     {
         // Mobile device types
-        if (!empty($this->device) && in_array($this->device, array(
-                DeviceParserAbstract::DEVICE_TYPE_FEATURE_PHONE,
-                DeviceParserAbstract::DEVICE_TYPE_SMARTPHONE,
-                DeviceParserAbstract::DEVICE_TYPE_TABLET,
-                DeviceParserAbstract::DEVICE_TYPE_PHABLET,
-                DeviceParserAbstract::DEVICE_TYPE_CAMERA,
-                DeviceParserAbstract::DEVICE_TYPE_PORTABLE_MEDIA_PAYER,
-            ))
+        if (!empty($this->device) && \in_array($this->device, [
```

Comment:
guess we actually do not even need the string cast in that case, as `$this->device` can either be a string or null. But even if it would be null or an empty string the in_array will return the expected value.